nifi-commons/nifi-vault-utils/src/main/java/org/apache/nifi/vault/config/VaultEnvironment.java

```diff
@@ -0,0 +1,179 @@
+
+    public VaultEnvironment(final VaultProperties vaultProperties) throws VaultConfigurationException {
+        Objects.requireNonNull(vaultProperties, "Vault Properties are required");
+        this.vaultProperties = vaultProperties;
+
+        Optional<Properties> authProperties = this.getAuthProperties(vaultProperties);
+        if (!authProperties.isPresent()) {
+            throw new VaultConfigurationException("Vault auth properties file is required");
+        } else {
+            this.authProperties = authProperties.get();
+        }
+    }
+
+    private Optional<Properties> getAuthProperties(VaultProperties properties) throws VaultConfigurationException {
+        if (properties.getAuthPropertiesFilename() == null) {
+            return Optional.empty();
+        }
+
+        File authPropertiesFile = Paths.get(properties.getAuthPropertiesFilename()).toFile();
+        if (!authPropertiesFile.exists()) {
+            return Optional.empty();
+        }
+
+        Properties authProperties = new Properties();
+        try (final FileReader reader = new FileReader(authPropertiesFile)) {
+            authProperties.load(reader);
+        } catch (final IOException e) {
+            final String message = String.format("Failed to read Vault auth properties [%s]", authPropertiesFile);
+            throw new VaultConfigurationException(message, e);
+        }
+        return Optional.of(authProperties);
+    }
+
+    @Override
+    public boolean containsProperty(String propertyName) {
+        return this.getProperty(propertyName) != null;
+    }
+
+    /**
+     * This maps the relevant Spring Vault properties to the configured properties from VaultProperties.
+     * @param key The Spring Vault property name (e.g., vault.uri)
+     * @return The configured property value
+     */
+    @Override
+    public String getProperty(String key) {
+        if (VAULT_URI.equals(key)) {
+            return vaultProperties.getUri();
+        } else if (VAULT_SSL_KEYSTORE.equals(key)) {
+            return vaultProperties.getKeystore();
+        } else if (VAULT_SSL_KEYSTORE_PASSWORD.equals(key)) {
+            return vaultProperties.getKeystorePassword();
+        } else if (VAULT_SSL_KEYSTORE_TYPE.equals(key)) {
+            return vaultProperties.getKeystoreType();
+        } else if (VAULT_SSL_TRUSTSTORE.equals(key)) {
+            return vaultProperties.getTruststore();
+        } else if (VAULT_SSL_TRUSTSTORE_PASSWORD.equals(key)) {
+            return vaultProperties.getTruststorePassword();
+        } else if (VAULT_SSL_TRUSTSTORE_TYPE.equals(key)) {
+            return vaultProperties.getTruststoreType();
+        } else if (VAULT_SSL_ENABLED_CIPHER_SUITES.equals(key)) {
+            return vaultProperties.getEnabledTlsCipherSuites();
+        } else if (VAULT_SSL_ENABLED_PROTOCOLS.equals(key)) {
+            return vaultProperties.getEnabledTlsProtocols();
+        } else {
+            return authProperties.getProperty(key);
+        }
```

Comment:
This seems a bit painful to maintain, as well as the other unimplemented methods.  Is it possible to use a different approach for translating the custom `VaultProperties` to an environment configuration?  Perhaps using annotating `VaultProperties` with Spring's `@ConfigurationProperties` and providing it to the custom Application Context?  Leveraging Spring's ability to support a variety of configuration options makes sense, but it would be helpful to evaluate other options for streamlining the approach.